spec/controllers/sessions_controller_spec.rb

```diff
@@ -1,5 +1,88 @@
 require 'rails_helper'
 
 RSpec.describe SessionsController, type: :controller do
+  describe "GET #create" do
+    before(:each) do
+      Subscription.create(
+        username: "Schwarzenegger", uid: "12044602" , provider: "twitter", avatar_url: "https://pbs.twimg.com/profile_images/665340796510466048/-nsoU1Q5.jpg"
+        )
 
+      Subscription.create(
+        username: "Schwarzenegger", uid: "12044602" , provider: "vimeo", avatar_url: "https://pbs.twimg.com/profile_images/665340796510466048/-nsoU1Q5.jpg"
+        )
+    end
+
+      context "when using twitter authentication" do
+        context "is successful" do
+          before { request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:twitter] }
+
+          it "redirects to home page" do
+            get :create, provider: :twitter
+            expect(response).to redirect_to root_path
+          end
+
+          it "creates a user" do
+            expect { get :create, provider: :twitter }.to change(User, :count).by(1)
+          end
+
+          it "assigns the @user var" do
+            get :create, provider: :twitter
+            expect(assigns(:user)).to be_an_instance_of User
+          end
+
+          it "assigns the session[:user_id]" do
+            get :create, provider: :twitter
+            expect(session[:user_id]).to eq assigns(:user).id
+          end
+        end
+
+        context "when the user has already signed up" do
+        before { request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:twitter] }
```

Comment:
Everything within the context of "when the user has already signed up" should be indented underneath it.